[compiled autograd] log aot id for CompiledFunctionBackward

xmfan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #133541
* #133148
* __->__ #133115

Partially addresses https://github.com/pytorch/pytorch/issues/132939. Adds the AOT ID after the CompiledFunctionBackward annotation in verbose compiled autograd logging

default (no change):
https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp8WCSLf/dedicated_log_torch_trace_xw3ktsi_.log/index.html

TORCH_LOGS="compiled_autograd_verbose":
https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp8WCSLf/dedicated_log_torch_trace_gsc9q_43.log/index.html

```python
# File: /data/users/xmfan/a/pytorch/torch/_dynamo/compiled_autograd.py:361 in set_node_origin, code: CompiledFunctionBackward1 (NodeCall 2)
clone: "f32[4]" = torch.ops.aten.clone.default(expand, memory_format = torch.contiguous_format);  expand = None
cos: "f32[4]" = torch.ops.aten.cos.default(getitem_1);  getitem_1 = None
mul: "f32[4]" = torch.ops.aten.mul.Tensor(clone, cos);  clone = cos = None

# File: /data/users/xmfan/a/pytorch/torch/_dynamo/compiled_autograd.py:361 in set_node_origin, code: CompiledFunctionBackward0 (NodeCall 3)
cos_1: "f32[4]" = torch.ops.aten.cos.default(getitem_2)
mul_1: "f32[4]" = torch.ops.aten.mul.Tensor(mul, cos_1);  mul = cos_1 = None
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

xmfan committed:
[compiled autograd] log aot id for CompiledFunctionBackward

[ghstack-poisoned]

xmfan committed:
Update on "[compiled autograd] log aot id for CompiledFunctionBackward"


Partially addresses https://github.com/pytorch/pytorch/issues/132939

default (no change):
https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp8WCSLf/dedicated_log_torch_trace_xw3ktsi_.log/index.html

TORCH_LOGS="compiled_autograd_verbose":
https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp8WCSLf/dedicated_log_torch_trace_gsc9q_43.log/index.html

```python
# File: /data/users/xmfan/a/pytorch/torch/_dynamo/compiled_autograd.py:361 in set_node_origin, code: CompiledFunctionBackward1 (NodeCall 2)
clone: "f32[4]" = torch.ops.aten.clone.default(expand, memory_format = torch.contiguous_format);  expand = None
cos: "f32[4]" = torch.ops.aten.cos.default(getitem_1);  getitem_1 = None
mul: "f32[4]" = torch.ops.aten.mul.Tensor(clone, cos);  clone = cos = None

# File: /data/users/xmfan/a/pytorch/torch/_dynamo/compiled_autograd.py:361 in set_node_origin, code: CompiledFunctionBackward0 (NodeCall 3)
cos_1: "f32[4]" = torch.ops.aten.cos.default(getitem_2)
mul_1: "f32[4]" = torch.ops.aten.mul.Tensor(mul, cos_1);  mul = cos_1 = None
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

yf225 commented on test/inductor/test_compiled_autograd.py:
```diff
@@ -2071,6 +2071,37 @@ def fn():
+            yield model[2].bias.grad
+
+        logs, ctx = logs_to_string(
+            torch._dynamo.compiled_autograd.__name__, "compiled_autograd_verbose"
```
Should we always enable it instead of only under `compiled_autograd_verbose`? I believe this feature would be very useful for debugging internal jobs in the future, and if this aot id logging is on by default then we don't need to ask user to go back and run the job again with verbose enabled. (Or alternatively we can also ask user to always turn on `compiled_autograd_verbose`)

xmfan replied:
yeah this one is not on the hot path so we can do that

bdhirsh replied:
also just thinking:

(1) do compiled autograd logs gets turned on for most jobs internally?

(2) is there anything else we can put into tlparse from compiled autograd that we aren't right now?

xmfan replied:
for 1, tlparse seems to have non-verbose logs by default

2, there's probably a bunch of cross-artifacts analysis we can do in tlparse. but i feel like adding logs to the compiled autograd artifacts is sufficient for the problems we're seeing rn

yf225 replied:
@bdhirsh another issue we saw is that when going from CA Dynamo graph to AOTAutograd joint graph, the source of each node in AOTAutograd is lost (i.e. we don't know which Dynamo graph FX node it comes from anymore). Curious is there a way to preserve this information?

bdhirsh replied:
hmm do you have an example, or a repro you can file?

I tried running the repro below, and when I the logs from `TORCH_LOGS="aot,+compiled_autograd"`, it seems like the stacktraces are lost when compiled autograd first gets a graph, before it sends it to AOTDispatcher. log output: P1528486054

code:
```
import torch


@torch.compile
def f(x):
    return torch.matmul(x, x).sin()

x = torch.randn(4, 4, requires_grad=True)
with torch._dynamo.utils.maybe_enable_compiled_autograd(True):
    out = f(x)
    out.sum().backward()

```

xmfan replied:
need to use compiled_autograd_verbose instead of +compiled_autograd (it's a separate artifact since artifacts don't support verbosity levels)

bdhirsh replied:
Filed here https://github.com/pytorch/pytorch/issues/133567, I'll take a look

bdhirsh replied:
This seems to fix it https://github.com/pytorch/pytorch/pull/133574

xmfan committed:
Update on "[compiled autograd] log aot id for CompiledFunctionBackward"


Partially addresses https://github.com/pytorch/pytorch/issues/132939. Adds the AOT ID after the CompiledFunctionBackward annotation in verbose compiled autograd logging

default (no change):
https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp8WCSLf/dedicated_log_torch_trace_xw3ktsi_.log/index.html

TORCH_LOGS="compiled_autograd_verbose":
https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp8WCSLf/dedicated_log_torch_trace_gsc9q_43.log/index.html

```python
# File: /data/users/xmfan/a/pytorch/torch/_dynamo/compiled_autograd.py:361 in set_node_origin, code: CompiledFunctionBackward1 (NodeCall 2)
clone: "f32[4]" = torch.ops.aten.clone.default(expand, memory_format = torch.contiguous_format);  expand = None
cos: "f32[4]" = torch.ops.aten.cos.default(getitem_1);  getitem_1 = None
mul: "f32[4]" = torch.ops.aten.mul.Tensor(clone, cos);  clone = cos = None

# File: /data/users/xmfan/a/pytorch/torch/_dynamo/compiled_autograd.py:361 in set_node_origin, code: CompiledFunctionBackward0 (NodeCall 3)
cos_1: "f32[4]" = torch.ops.aten.cos.default(getitem_2)
mul_1: "f32[4]" = torch.ops.aten.mul.Tensor(mul, cos_1);  mul = cos_1 = None
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]
